### PR TITLE
feat: Role choice when Signing Up

### LINKED ISSUE #38: #12: Add role choice on signup process
1) Read guideline [here](https://gist.github.com/iamnovichek/801ac395a06a7d8b04b022084ace684d)
2) Add 2 existing roles (from `tracker/choices.py`) to `User` model as `role` field of type `CharField` and set `contributor` as a default one
3) Update migrations
4) Add the possibility to choose the role on signup (Droplist)

### PR DESCRIPTION
**Pull request by Goodnessukaigwe**
close #38

### CODE REVIEW

File: tracker/forms.py [hunk 1]
```diff
@@ -2,7 +2,9 @@
 from django.core.exceptions import ValidationError
 from django.db import transaction
 
-from .models import CustomUser
+from tracker.choices import Roles
```
Comment (iamnovichek) [inline tracker/forms.py]: Relative import works as well (`.choices`)

File: tracker/forms.py [hunk 2]
```diff
@@ -29,6 +31,7 @@ class SignUpForm(forms.ModelForm):
     email = forms.EmailField()
     password = forms.CharField(widget=forms.PasswordInput())
     confirm_password = forms.CharField(widget=forms.PasswordInput())
+    role = forms.ModelChoiceField(queryset=CustomUser.objects.all())
```
Comment (iamnovichek) [inline tracker/forms.py]: Please use `role = forms.ChoiceField(choices=Roles.choices)`

File: tracker/models.py [hunk 3]
```diff
@@ -74,6 +75,11 @@ class CustomUser(AbstractModel, AbstractBaseUser):
     )
     is_active = models.BooleanField(default=True)
     is_admin = models.BooleanField(default=False)
+    role = models.CharField(
+        max_length=11,
```
Comment (iamnovichek) [inline tracker/models.py]: Please move this magic number to `values.py`

File: tracker/models.py [hunk 4]
```diff
@@ -74,6 +75,11 @@ class CustomUser(AbstractModel, AbstractBaseUser):
...
+        choices=Roles,
```
Comment (iamnovichek) [inline tracker/models.py]: Please use `Roles.choices`

File: tracker/templates/signup.html [hunk 5]
```diff
@@ -65,6 +65,13 @@
                                 <label for="id_confirm_password" class="required">Confirm password:</label>
                                 <input type="password" name="confirm_password" required id="id_password">
                             </div>
+                            <div class="form-row">
+                                <label for="role" class="required">Role</label>
+                                <select id="role" name="role">
+                                    <option value="contributor">Contributor</option>
```
Comment (iamnovichek) [inline tracker/templates/signup.html]: Use `for` loop. What if we will add move roles in the future?

**Review by iamnovichek — CHANGES_REQUESTED** [review]

File: tracker/values.py [hunk 6]
```diff
@@ -10,6 +10,7 @@
 PULLS_REVIEWS_URL = (
     "https://api.github.com/repos/{owner}/{repo}/pulls/{pull_number}/reviews"
 )
+ROLES_MAX_FIELD = 11
```
Comment (iamnovichek) [inline tracker/values.py]: Bad constant name. Pls move it to dataclass as a class variable

**Review by iamnovichek — CHANGES_REQUESTED** [review]
Please provide some screenshots to prove your code works

**Comment by iamnovichek** [comment]
@Goodnessukaigwe Pls sync with main

File: tracker/models.py [hunk 7]
```diff
@@ -38,27 +39,30 @@ def create_user(self, email: str = None, password: str = None) -> "CustomUser":
...
+    def create_superuser(self, email: str = None, password: str = None, role: str = None) -> "CustomUser":
         """
         Creates a new superuser
         :param email: str = None
         :param password: str = None
+        :param role: str = None
         :return: CustomUser
         """
-        user = self.create_user(email, password)
+        user = self.create_user(email, password, role)
```
Comment (iamnovichek) [inline tracker/models.py:52]: it won't work because `create_user` does not accept the third argument

File: tracker/models.py [hunk 8]
```diff
@@ -38,27 +39,30 @@ def create_user(self, email: str = None, password: str = None) -> "CustomUser":
...
 
         user.is_admin = True
         user.save(update_fields=["is_admin", "is_active"])
 
         return user
 
 
+ROLE_MAX_CHARACTER_LENGTH = 11
```
Comment (iamnovichek) [inline tracker/models.py]: Pls move it to `values.py`

**Review by iamnovichek — CHANGES_REQUESTED** [review]
Pls provide also some screenshots to prove that it works

**Review by iamnovichek — CHANGES_REQUESTED** [review]
Pls provide some screenshots to prove that it works

**Comment by iamnovichek** [comment]
@Goodnessukaigwe Pls resolve your merge conflicts

File: tracker/models.py [hunk 9]
```diff
@@ -1,13 +1,14 @@
...
 from django.conf import settings
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager
 from django.core.exceptions import ValidationError
 from django.db import models
 from django_celery_beat.models import IntervalSchedule, PeriodicTask
 
 from shared.models import AbstractModel
 
-from .values import DefaultModelValues
+from .choices import Roles
```
Comment (djeck1432) [inline tracker/models.py]: don't use `relative import`

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Pull request merged**